curvefs/src/client/fuse_client.cpp

```diff
@@ -678,6 +812,246 @@ CURVEFS_ERROR FuseClient::FuseOpSetAttr(fuse_req_t req, fuse_ino_t ino,
+    if (CURVEFS_ERROR::OK != ret) {
+        LOG(ERROR) << "ListDentry failed, inodeId = " << ino
+                    << ", limit = " << limit << ", onlyDir = false";
+        return ret;
+    }
+    for (const auto &it : dentryList) {
+        if (it.type() == FsFileType::TYPE_DIRECTORY) {
+                iStack.emplace(it.inodeid());
+        }
+        inodeIds.emplace(it.inodeid());
+    }
+    ret = inodeManager_->BatchGetInodeAttr(inodeIds, &attrs);
+    if (ret == CURVEFS_ERROR::OK) {
+        uint64_t files = 0;
+        uint64_t subdirs = 0;
+        uint64_t entries = 0;
+        uint64_t fbytes = 0;
+        for (const auto &it : attrs) {
+            if (it.type() == FsFileType::TYPE_DIRECTORY) {
+                subdirs++;
+            } else {
+                files++;
+            }
+            entries++;
+            fbytes += it.length();
+        }
+        AddUllStringToFirst(
+            &(inode->mutable_xattr()->find(XATTRFILES)->second),
+                std::to_string(files), true);
+        AddUllStringToFirst(
+            &(inode->mutable_xattr()->find(XATTRSUBDIRS)->second),
+                std::to_string(subdirs), true);
+        AddUllStringToFirst(
+            &(inode->mutable_xattr()->find(XATTRENTRIES)->second),
+                std::to_string(entries), true);
+        AddUllStringToFirst(
+            &(inode->mutable_xattr()->find(XATTRFBYTES)->second),
+                std::to_string(fbytes), true);
+    } else {
+        return ret;
+    }
+
+    // all layer
+    uint64_t subdir = 0;
+    while (!iStack.empty()) {
+        ino = iStack.top();
+        iStack.pop();
+        std::list<Dentry> dentryList;
+        ret = dentryManager_->ListDentry(ino, &dentryList, limit, false);
+        if (CURVEFS_ERROR::OK != ret) {
+            LOG(ERROR) << "ListDentry failed, inodeId = " << ino
+                       << ", limit = " << limit << ", onlyDir = false";
+            return ret;
+        }
+        for (const auto &it : dentryList) {
+            if (it.type() == FsFileType::TYPE_DIRECTORY) {
+                iStack.emplace(it.inodeid());
+                subdir++;
+            }
+            inodeIds.emplace(it.inodeid());
+        }
+    }
+
+    VLOG(1) << "RcalculateSumInfo subdir_num = " << subdir
+            << ", entries_num = " << inodeIds.size();
+    attrs.clear();
+    ret = inodeManager_->BatchGetInodeAttr(inodeIds, &attrs);
+    if (ret == CURVEFS_ERROR::OK) {
+        uint64_t rfiles = 0;
+        uint64_t rsubdirs = 0;
+        uint64_t rentries = 0;
+        uint64_t rfbytes = 0;
+        for (const auto &it : attrs) {
+            if (it.type() == FsFileType::TYPE_DIRECTORY) {
+                rsubdirs++;
+            } else {
+                rfiles++;
+            }
+            rentries++;
+            rfbytes += it.length();
+        }
+        inode->mutable_xattr()->insert({XATTRRFILES, std::to_string(rfiles)});
```

Comment:
if the key already exist, this can success?

Reply:
> if the key already exist, this can success?

The curve.dir.r* have not recorded in dir xattr. Just pass the calculated value througth the temp inode xattr.